### ISSUE TITLE
Material UI prop "InputProps" is not supported in Formik Material UI select wrapper component

### ISSUE THREAD

**Issue by PeterWolf91** [stackworx/formik-material-ui#48]
Hello,
first of all thank you for the work on this project 🥇 . With TextFields everything is working fine 👍 .

I am currently trying to use the Select wrapper component of this repository, but the documentation of this feature is not so much.  At least one prop which the official Material UI component TextField, in context of select, supports is currently not working. This acccounts especially for the prop "InputProps" which makes the component able to display icons inside the select. In addition, I don't understand why the label is not being displayed. Before I created this issue, I was looking at your stories and source code but couldn't determine what I might have done wrong.

To illustrate , I created a short example on CodeSandBox:

- https://codesandbox.io/s/l76n9n6nvq

I would appreciate a short feedback. Thanks in advance.

**Comment by cliedeman** [comment]
Hi there. 

Unfortunately Select does not have a Input Label included in its api.

Snippet from the samples [here](https://material-ui.com/demos/selects/):

```
       <FormControl className={classes.formControl} disabled>
          <InputLabel htmlFor="name-disabled">Name</InputLabel>
          <Select
            value={this.state.name}
            onChange={this.handleChange}
            input={<Input name="name" id="name-disabled" />}
          >
            <MenuItem value="">
              <em>None</em>
            </MenuItem>
            <MenuItem value="hai">Hai</MenuItem>
            <MenuItem value="olivier">Olivier</MenuItem>
            <MenuItem value="kevin">Kevin</MenuItem>
          </Select>
          <FormHelperText>Disabled</FormHelperText>
        </FormControl>
```

I adapted your [sandbox](https://codesandbox.io/s/8x368y761j).

Ciaran

**Comment by PeterWolf91** [comment]
Hello Ciaran,

thank you for your reply. In the adapted sandbox, the icon is not being displayed for me. Was it working for you?

While I was waiting yesterday for an answer by someone, I came up with the following version ([sandbox](https://codesandbox.io/s/8zwnrm2oy0)):

```jsx
<FormControl fullWidth required>
    <InputLabel
        htmlFor="group"
        className={classes.special}
    >
        Group
    </InputLabel>
    <Field
        name="group"
        component={Select}
        startAdornment={
            <InputAdornment position="start">
                <GroupIcon />
            </InputAdornment>
        }
    >
      <MenuItem value="sea">Sea</MenuItem>
      <MenuItem value="land">Land</MenuItem>
      <MenuItem value="air">Air</MenuItem>
    </Field>
</FormControl>
```

The problems with this version is that the field still not displays any error message or marks it as required when you click on "Insert data" directly. Furthermore, the label is displayed over the icon in initial state. With a css class and left padding of 30px, you can create a workaround which then looks odd when a values has been selected. Any idea?

Thanks in advance.

**Comment by cliedeman** [comment]
Select is pretty barebones compared to TextField. You will have to Add FormHelperText [yourself](https://codesandbox.io/s/p3pwyykwqj).

As for the styling issue it happens with [plain material ui](https://codesandbox.io/s/pwz7jzrwn7). So that is either a bug in material ui or something you have to manually style

Ciaran

**Comment by PeterWolf91** [comment]
Hi Ciaran,

thank you for your feedback. With your version I get at least a message to the user but the select field does not get coloured red. With this tradeoff I can live. Another possibility to achieve this desired behaviour would be to replace this part:

```jsx
<Field
  name="group"
  render={({ field, form }) => {
    const error = getIn(form.errors, field.name);
    return (
      !!error && (
        <FormHelperText error={true}>
          {error}
        </FormHelperText>
      )
    );
  }}
/>
```

with this part:

```jsx
{errors.actionArea && (
  <FormHelperText error>
      {errors.actionArea}
  </FormHelperText>
)}
```

But to use the second approach, you have to pass the **errors** object to the render prop of Formik like this:

```jsx
render={({ errors, submitForm, isSubmitting }) => (
```

**Comment by cliedeman** [comment]
I looked into this a bit more and found there is a select variation of TextField that was not working correctly. I have added an example [here](https://stackworx.github.io/formik-material-ui/?selectedKind=Formik&selectedStory=Text%20Field). It may work for your use case if you are not doing a multi-select. The fix was also published in v0.0.16

Ciaran

**Comment by PeterWolf91** [comment]
Hi,

thanks for taking your time. I tried out your example but as soon as I add an adornment like this to your example it lacks the "With Select" label:

```jsx
<Field
  type="text"
  name="select"
  label="With Select"
  select
  SelectProps={{
    MenuProps: {
      className: classes.menu,
    },
  }}
  helperText="Please select Range"
  margin="normal"
  component={TextField}
  InputProps={{
      startAdornment: (
          <InputAdornment position="start">
              <MaterialUiIcon />
          </InputAdornment>
      ),
  }}
>
```
